modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java

```diff
@@ -164,15 +181,27 @@ private boolean versionUsedInAnyBranch( final Repository repository, final NodeV
         return false;
     }
 
-    private NodeVersionQuery createQuery( final VacuumTaskParams params )
+    private NodeVersionQuery createQuery( final NodeVersionId lastVersionId )
     {
-        final Instant since = Instant.now().minusMillis( params.getAgeThreshold() );
+        final NodeVersionQuery.Builder builder = NodeVersionQuery.create();
+
+        if ( lastVersionId != null )
+        {
+            final RangeFilter versionIdFilter = RangeFilter.create().
+                fieldName( VersionIndexPath.VERSION_ID.getPath() ).
+                from( ValueFactory.newString( lastVersionId.toString() ) ).
```

Comment:
"from" is inclusive and may end up in infinite loop. In 7.2 Range filter is extended to support includeLower.

Reply:
comment added.